### PR TITLE
Private root certificates

### PR DESCRIPTION
**Pull request by de31a2**
#### What this PR does / why we need it:
This PR enables adding additional volumes and volumemounts to add private CA root certificates. / It is needed if the private location needs to check services with a with a private CAs certificate

#### Special notes for your reviewer:
this solution follows the way described in https://docs.datadoghq.com/synthetics/private_locations/configuration/#private-root-certificates

#### Checklist
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] Chart Version bumped
- [x] `CHANGELOG.md` has beed updated
- [x] Variables are documented in the `README.md`

### CODE REVIEW

**Comment by de31a2** [comment]
Rebased onto the current main branch.

**Comment by brycekahle** [comment]
@de31a2 This PR seems to include a bunch of unrelated commits (31 according to Github). Maybe a fresh rebase is in order?

**Comment by de31a2** [comment]
> @de31a2 This PR seems to include a bunch of unrelated commits (31 according to Github). Maybe a fresh rebase is in order?

Yes. I just rebased to a this branch

**Comment by m-rousse** [comment]
Hi @de31a2,

Looking at the commits, it seems your modifications were doubled, which seem to disturb github a bit:
- ab7d43b, b921f85 and 97d5fbb seems to be very close/almost identical to
- 8c1c867, 3b5731f and c4ff454

I would suggest resetting on `main` and re-applying your changes in your branch to have it in one commit, I would do:
```
git diff origin/main -p > /tmp/patch
git reset --hard origin/main
patch -p0 < /tmp/patch
git commit -m '[synthetics] add private root certificates support'
git push
```

**Comment by m-rousse** [comment]
Hi @de31a2,

My apologies for not getting back to you about this, I missed the notification when the branch was updated.
I opened a PR on your branch to merge `main` into it and fix conflicts: https://github.com/kiongroup/tdi-datadog-helm-charts/pull/3

Once you've reviewed it/merged it, I'll be ready to approve this one and ask for a merge.

File: charts/synthetics-private-location/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Datadog changelog
 
+### 0.13.0
+
+* added extra mount (`extraVolumes` and `extraVolumeMounts` ) for supporting private root CA certificates. As described in https://docs.datadoghq.com/synthetics/private_locations/configuration/#private-root-certificates
```
Comment (badouralix) [inline charts/synthetics-private-location/CHANGELOG.md:5]:
```suggestion
* Add extra mount (`extraVolumes` and `extraVolumeMounts`) for supporting private root CA certificates as described in <https://docs.datadoghq.com/synthetics/private_locations/configuration/#private-root-certificates>.
```

File: charts/synthetics-private-location/templates/deployment.yaml [hunk 2]
```diff
@@ -46,6 +46,9 @@ spec:
           volumeMounts:
           - mountPath: /etc/datadog
             name: worker-config
+          {{- if .Values.extraVolumeMounts }}
+            {{- toYaml .Values.extraVolumeMounts | nindent 10 }}
+            {{- end }}
```
Comment (badouralix) [inline charts/synthetics-private-location/templates/deployment.yaml:51]:
```suggestion
          {{- toYaml .Values.extraVolumeMounts | nindent 10 }}
          {{- end }}
```

File: charts/synthetics-private-location/templates/deployment.yaml [hunk 3]
```diff
@@ -81,3 +84,6 @@ spec:
           secretName: {{ include "synthetics-private-location.fullname" . }}-config
         {{- end }}
         {{- end }}
+        {{- if .Values.extraVolumes }}
+        {{- toYaml .Values.extraVolumes | nindent 6 }}
+        {{- end }}
```
Comment (badouralix) [inline charts/synthetics-private-location/templates/deployment.yaml:89]:
```suggestion
      {{- if .Values.extraVolumes }}
      {{- toYaml .Values.extraVolumes | nindent 6 }}
      {{- end }}
```

File: charts/synthetics-private-location/values.yaml [hunk 4]
```diff
@@ -26,6 +26,23 @@ serviceAccount:
   # serviceAccount.name -- The name of the service account to use. If not set name is generated using the fullname template
   name: ""
 
+# Add your custom CA Root certificate in PEM format
+# first create the ConfigMap containing the PEM files
+# than the content of this CM will be added as files to /etc/datadog/certs/
```
Comment (badouralix) [inline charts/synthetics-private-location/values.yaml:31]:
```suggestion
# Create a ConfigMap containing the PEM files of your custom CA Root certificate
# Then add it as an extra volume mounted on /etc/datadog/certs/
```

**Review by badouralix — CHANGES_REQUESTED** [review]
Hi @de31a2, thank you for your contribution. Overall lgtm ! I have added a few nitpicks to fix the linting errors. Would it be possible to take care of them before we go ahead and merge this pull request ?

**Pull request closed without merging**